ANMN/LTSP/TSaggregator/copyDataset.py

```diff
@@ -1,66 +1,130 @@
+             'http://thredds.aodn.org.au/thredds/dodsC/IMOS/ANMN/NRS/NRSROT/Temperature/IMOS_ANMN-NRS_TZ_20171124T080000Z_NRSROT_FV01_NRSROT-1712-SBE39-43_END-20180409T060000Z_C-20180503T020214Z.nc',
+             'http://thredds.aodn.org.au/thredds/dodsC/IMOS/ANMN/NRS/NRSROT/Temperature/IMOS_ANMN-NRS_TZ_20180406T080000Z_NRSROT_FV01_NRSROT-1804-SBE39-27_END-20180817T023000Z_C-20180820T010304Z.nc',
+             'http://thredds.aodn.org.au/thredds/dodsC/IMOS/ANMN/NRS/NRSROT/Temperature/IMOS_ANMN-NRS_TZ_20180406T080000Z_NRSROT_FV01_NRSROT-1804-SBE39-43_END-20180817T025000Z_C-20180820T010304Z.nc',
+             'http://thredds.aodn.org.au/thredds/dodsC/IMOS/ANMN/NRS/NRSROT/Temperature/IMOS_ANMN-NRS_TZ_20180816T080000Z_NRSROT_FV01_NRSROT-1808-SBE39-27_END-20181214T034000Z_C-20190402T065832Z.nc',
+             'http://thredds.aodn.org.au/thredds/dodsC/IMOS/ANMN/NRS/NRSROT/Temperature/IMOS_ANMN-NRS_TZ_20180816T080000Z_NRSROT_FV01_NRSROT-1808-SBE39-43_END-20181214T030000Z_C-20190402T065833Z.nc']
+else:
 
-files = []
-varToAgg = []
+    # print(type(args.site))
+    # sys.exit()
 
-if len(sys.argv) > 1:
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-v', action='append', dest='var', help='variable to include in output file (defaults to all)')
-    parser.add_argument('-f', dest='filelist', help='read file names from file')
-    parser.add_argument('file', nargs='*', help='input file name')
-    args = parser.parse_args()
+    if args.var not in var_names_dict.keys() or isinstance(args.var, type(None)):
+        sys.exit('ERROR: invalid variable name.')
 
-    if not isinstance(args.filelist, type(None)):
-        with open(args.filelist, "r") as ins:
-            for line in ins:
-                print(line)
-                files.append(line.strip())
+    if isinstance(args.site, type(None)):
+        sys.exit('ERROR: missing site.')
 
-    if len(args.file):
-        # files = args.file
-        for fn in args.file:
-            files.extend(glob.glob(fn))
+    print('Concatenating %s from %s since %s thru %s' % (args.var, args.site, args.timeStart, args.timeEnd))
+    var_to_agg = [args.var]
 
-    varToAgg = args.var
-else:
-    files=["EAC-2000/IMOS_ABOS-DA_STZ_20150515T000001Z_EAC2000_FV01_EAC2000-2016-SBE37SMP-140_END-20161110T221930Z_C-20170703T055824Z.nc",
-           "EAC-2000/IMOS_ABOS-DA_STZ_20150515T000001Z_EAC2000_FV01_EAC2000-2016-SBE37SMP-205_END-20161110T224850Z_C-20170703T055825Z.nc"]
+    # get the file names and attr from the geoserver
+    # Only FV01 files
+    print('Getting the file names...')
+    url = "http://geoserver-123.aodn.org.au/geoserver/ows?typeName=moorings_all_map&SERVICE=WFS&REQUEST=GetFeature&VERSION=1.0.0&outputFormat=csv&CQL_FILTER=(file_version='1'%20AND%20realtime=FALSE%20AND%20strToLowerCase(feature_type)='timeseries')"
```

Comment:
Actually, filtering by `feature_type` is not a good idea, because unfortunately the value of this attribute is not consistent across the whole collection. In particular, for ADCP files, it could be null, or 'timeSeries' or 'timeSeriesProfile', depending on the version of the Toolbox that processed the file.

Again, I think selecting the input files list should be treated separately from the aggregation. I suggest putting any code handling that selection in a separate function that outputs a simple list of URLs. This should even be in a separate .py file. (It could actually be useful in its own right, for other purposes too.)

Reply:
I created a new stand alone function for extracting the url list form the geoserver. However the current version of aggregator is not using it already. But I think it is OK to distributed the current version.

Reply:
Also I've included a new filtering criteria that leaves out urls with  the data_category=='Velocity'